LIBITD-2532. Created API's check view

aatishgupta25 opened the pull request:
Made a new url pattern for check and made a new view to handle the query parameters and provide appropriate response with different status codes (200, 400, 404).

https://umd-dit.atlassian.net/browse/LIBITD-2532

Commit:
LIBITD-2532. Created API's check view

Made a new url pattern for check and made a new view
to handle the query parameters and provide appropriate
response with different status codes (200, 400, 404).

https://umd-dit.atlassian.net/browse/LIBITD-2532

peichman-umd commented on src/ipmanager/api/views.py:
```diff
@@ -43,4 +44,55 @@ def get(self, request, group_key):
+            return JsonResponse(body, status=404)
+        
+
+class APICheckView(View):
```
To align with the other view classes, this class should be named `CheckView` (drop the `API` portion of the name).

peichman-umd commented on src/ipmanager/api/views.py:
```diff
@@ -43,4 +44,55 @@ def get(self, request, group_key):
+    group_keys = ["A", "B", "C", "D"]
+
+    # Using query_string (unparsed) and to create Query Dict object to extrack parameters
+    dict = QueryDict(query_string = request.META['QUERY_STRING'], mutable=False, encoding=None)
```
You don't need to parse the incoming query string yourself: Django does it for you, and puts the results into the `GET` attribute of the `request`.

```python
if 'ip' not in request.GET:
    return JSONResponse(...)

ip = request.GET['ip']
```

peichman-umd commented on src/ipmanager/api/views.py:
```diff
@@ -43,4 +44,55 @@ def get(self, request, group_key):
+    
+    # Returning 400 error if wrong parameters or ip not present
+    if 'ip' not in dict:
+      return JsonResponse({"title": "Unparseable IP address", 
```
The error you are responding to here is a missing parameter, not an unparseable IP address, so the `title` and `detail` should reflect that.

peichman-umd commented on src/ipmanager/api/views.py:
```diff
@@ -43,4 +44,55 @@ def get(self, request, group_key):
+                             "group":{"@id":"https://ipmanager-test.lib.umd.edu/groups/henson20001",
+                                      "key":group,
+                                      "name":"Henson Collection"},
+                             "ip_address":
```
Formatting: `ip` should be on the same line as `"ip_address":`

peichman-umd reviewed (changes requested):
The logic looks good. Just a handful of improvements that I have noted inline.

peichman-umd commented on src/ipmanager/api/views.py:
```diff
@@ -43,4 +44,55 @@ def get(self, request, group_key):
+                           "ip":ip,
+                           # Demo-only data, with actual data iterate through all group_keys and make url and parse response accordingly
+                           "checks":[{"@id":"https://ipmanager-test.lib.umd.edu/check?group=Henson00001&ip=127.0.0.1",
+                                      "group":{"@id":"ttps://ipmanager-test.lib.umd.edu/groups/Henson00001",
```
I know this is just placeholder data, but you are missing an "h" at the beginning of the group `@id`.

Commit:
LIBITD-2532. Made changes to CheckView

Now checking for both unparseable and missing ip address.
Fixed Indentation for ip field. Renamed APICheckView to
CheckView. Removed the use of QUERY_STRING and used request.GET
instead.

https://umd-dit.atlassian.net/browse/LIBITD-2532

aatishgupta25 committed:
Merge pull request #1 from aatishgupta25/feature/check-api-test

LIBITD-2532. Made changes to CheckView

peichman-umd commented on src/ipmanager/api/views.py:
```diff
@@ -47,45 +47,47 @@ def get(self, request, group_key):
+        IPv4Address(ip)
+    except AddressValueError:
+        return JsonResponse({"title": "Unparseable IP address", 
+                             "detail": "The value 'invalid_ip' you provided in the 'ip' query parameter cannot be " + 
```
You should insert the actual submitted value in place of "invalid_ip". A good way to do this is using Python [formatted string literals](https://docs.python.org/3/tutorial/inputoutput.html#tut-f-strings) (a.k.a., "f-strings"):

```python
f"The value '{ip}' you provided in the 'ip' query parameter cannot be "
```

peichman-umd reviewed (changes requested):
One minor change to the error message for invalid IP address.

Commit:
LIBITD-2532. Updated CheckView

Update CheckView to use f-string to print error message.

https://umd-dit.atlassian.net/browse/LIBITD-2532

aatishgupta25 committed:
Merge pull request #2 from aatishgupta25/feature/check-api-test

LIBITD-2532. Updated CheckView
